llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -2674,10 +2674,41 @@ Instruction *InstCombinerImpl::foldICmpShrConstant(ICmpInst &Cmp,
 Instruction *InstCombinerImpl::foldICmpSRemConstant(ICmpInst &Cmp,
                                                     BinaryOperator *SRem,
                                                     const APInt &C) {
+  const ICmpInst::Predicate Pred = Cmp.getPredicate();
+  if (Pred == ICmpInst::ICMP_UGT || Pred == ICmpInst::ICMP_ULT) {
+    // Canonicalize unsigned predicates to signed:
+    // (X % DivisorC) ugt C -> (X % DivisorC) slt 0
+    //   iff (C slt 0 ? ~C : C) uge abs(DivisorC)-1
+    // (X % DivisorC) ult C+1 -> (X % DivisorC) sgt -1
+    //   iff (C+1 slt 0 ? ~C : C) uge abs(DivisorC)-1
```

Comment:
nit: For the comment can you be consistent w/ either ~C syntax or ~IR syntax i.e


`(X % DivisorC) ugt C -> (X % DivisorC) slt 0)` -> `(icmp ugt (srem X, DivisorC), C) -> (icmp slt (srem X, DivisorC), 0)` or `(X s% DivisorC) u> C` -> `(X s% DivisorC) s< 0`

Reply:
I was trying to copy the style of the nearby code, but I certainly wanted to write `s%` and `s<` so I'll do that.